src/rust/lib/graph-editor/src/lib.rs

```diff
@@ -287,6 +286,7 @@ pub struct FrpInputs {
     pub translate_selected_nodes       : frp::Source<Position>,
     pub cycle_visualization            : frp::Source<NodeId>,
     pub set_visualization              : frp::Source<(NodeId,Option<Visualization>)>,
+    pub register_visualisation_source  : frp::Source<Option<Rc<dyn visualization::Factory>>>,
```

Comment:
Inconsistency - source and factory. After reviewing almost all the code I believe this name should be class and we should have here `register_visualisation_class`. Moreover, frp types needs to be simple - please create newtype for every trait object you have!